scripts/subscribeEventProof.js

```diff
@@ -1,110 +1,92 @@
+async function  extractNewValidators(api, blockHash) {
+    const notaryKeys = await api.query.ethBridge.nextNotaryKeys.at(blockHash);
+    const newValidators = notaryKeys.map((notaryKey) => {
+        if (notaryKey.toString() === IGNORE_KEY) return '0x0000000000000000000000000000000000000000';
+        let decompressedPk = ethers.utils.computePublicKey(notaryKey);
+        let h = ethers.utils.keccak256('0x' + decompressedPk.slice(4));
+        return '0x' + h.slice(26)
+    });
+    return newValidators;
 }
 
-function getValidatorAddedToBridge(bridge) {
-    let lastValidatorsSet = {};
+async function updateLastEventProcessed(eventId, blockHash) {
+    const filter = {};
+    const update = { eventId, blockHash };
+    const options = { upsert: true, new: true, setDefaultsOnInsert: true }; // create new if record does not exist, else update
+    await EventProcessed.updateOne(filter, update, options);
+}
 
-    const filter = {
-        address: bridge.address,
-        fromBlock: 0,
-        topics: [
-            ethers.utils.id("SetValidators(address[],uint256,uint32)")
-        ]
-    }
+// Submit the event proof on Ethereum Bridge contract
+async function getEventPoofAndSubmit(api, eventId, bridge, txExecutor, newValidatorSetId, blockHash) {
+    try {
 
-    ethers.provider.getLogs(filter).then((result) => {
-        if (result.length > 0) {
-            const abiCoder = ethers.utils.defaultAbiCoder;
-            const decodedResponse = abiCoder.decode(['address[]', 'uint256', 'uint32'], result[result.length - 1].data);
-            lastValidatorsSet.validators = decodedResponse[0].map(validator => validator.toLowerCase());
-            lastValidatorsSet.id = decodedResponse[2].toString();
-            logger.info(`Previous Validator set in bridge is: ${JSON.stringify(lastValidatorsSet)}`);
+        const eventExistsOnEth = await bridge.eventIds(eventId.toString());
+        const eventProof = await withTimeout(api.derive.ethBridge.eventProof(eventId), 20000);
+        if (eventProof && !eventExistsOnEth) {
+            const newValidators = await extractNewValidators(api, blockHash);
+            logger.info(`Sending setValidators tx with the account: ${txExecutor.address}`);
+            logger.info(`Parameters :::`);
+            logger.info(`newValidators:${newValidators}`);
+            logger.info(`newValidatorSetId: ${newValidatorSetId}`);
+            logger.info(`event proof::${JSON.stringify(eventProof)}`);
+            const proof = {eventId: eventProof.eventId, validatorSetId: eventProof.validatorSetId, r: eventProof.r, s: eventProof.s, v: eventProof.v};
+            const gasEstimated = await bridge.estimateGas.setValidators(newValidators, newValidatorSetId, proof, {gasLimit: 500000});
+            logger.info(JSON.stringify(await bridge.setValidators(newValidators, newValidatorSetId, proof, {gasLimit: 500000})));
```

Comment:
we don't want this to fail if the gas goes over
```suggestion
            logger.info(JSON.stringify(await bridge.setValidators(newValidators, newValidatorSetId, proof, {gasLimit: estimatedGas + buffer })));
```